widget_driver/README.md

```diff
@@ -337,7 +337,46 @@ Easy...
     ```
 
 2. Then we just run the generator like we did before...
-3. After that we just need to hand the variable over to the `DriverProvider` and that's it. 🥳
+3. We should get a compiler warning in the generated file. To resolve that we just have to add the generated mixin `$<YourDriver>ProvidedPropertiesMixin` to our driver like this:
+
+    ```dart
+    @GenerateTestDriver()
+      class CoffeeDetailPageDriver extends WidgetDriver with $CoffeeDetailPageDriverProvidedPropertiesMixin {
+        ...
+      }
+    ```
+
+4. This requires us to override `updateDriverProvidedProperties(...)` which gets called on state updates. That way we can respond to new values to our provided properties. (Technical Note: This is because the Driver does not get rebuilt on state updates. And a call to `notifyWidget()` is not necessary, this function gets called before the widget shows the new data.)
```

Comment:
It might be dangerous to use the word `state changes` here.. Since the state is actually not really changing..
A state change sounds a bit like someone calling `setState` on the state.

Implementation details => The driver lives inside the state of a StatefulWidget.
The widget which belongs to the state has changed. And the properties come from that widget.
So it is more like the widget which provided the properties have changed.

So to be more acurate and not lead to confusion. maybe we phrase this:
```
This requires us to override `updateDriverProvidedProperties(...)` which gets called on DrivableWidgets updates.
That way we can respond to new values to our provided properties coming from the DrivableWidget.
```

Reply:
I changed it
Does it work for you this way.
For me at least, state change is that the widget got new data to display, so the widget rebuilt (which would be somebody called `setState` somewhere)